openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1133,6 +1133,38 @@ def delete_library_block(usage_key, remove_from_parent=True):
         )
 
 
+def restore_library_block(usage_key):
+    """
+    Restore the specified library block.
+    """
+    component = get_component_from_usage_key(usage_key)
+    library_key = usage_key.context_key
+    affected_collections = authoring_api.get_entity_collections(component.learning_package_id, component.key)
+
+    # Set draft version back to the latest available component version id.
+    authoring_api.set_draft_version(component.pk, component.versioning.latest.pk)
```

Comment:
We can add a helper method in openedx-learning similar to [soft_delete_draft](https://github.com/openedx/openedx-learning/blob/ed7b9e4162b9e280607410ece9d39786e5eedd2c/openedx_learning/apps/authoring/publishing/api.py#L410-L420) if required.